src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState } from 'react';
+
+import * as RadixCheckbox from '@radix-ui/react-checkbox';
+import { IconCheckBoxCheck } from '../Icons';
+
+import classNames from 'classnames';
+import styles from './Checkbox.module.scss';
+const cx = classNames.bind(styles);
+
+export interface CheckboxProps {
+  className?: string;
+  name: string;
+  text?: string;
+  link?: string;
```

Comment:
I think `link` is slightly confusing here, because it's not obvious if `link` is the URL to link to or the link text. Perhaps we need a `Link` type, `interface Link { text: string, href: string }` or `{ text: string, onClick: () => void }`?

Reply:
The Link may not be always URL right. Maybe we can do a routing redirect or external link we want to open in the same tab or a different tab. So, I added it as an event we can use accordingly where the component is getting created.